Senec: add option to enable/disable fetching STATISTIC API

Soph opened the pull request:
This is a follow up to https://github.com/evcc-io/evcc/pull/9492

While the `schema` part works nicely with my updated Senec the newly added querying of the `{"STATISTIC":{"LIVE_GRID_IMPORT":""}}` properties wasn't working with my Senec V2.1. I can only assume this is something added in V3. 

I added a new parameter `live_stats` to enable fetching the energy values but it defaults to `false`. Happy to give this a better name. An option could be to use a `version: ` param and then only enable it for version 3 and up. 

I tested this locally with `live_stats: true` (fails), `live_stats: false` (works) and with out `live_stats` at all (works too)

/cc @premultiply

Soph committed:
making energy tracking optional

Commit:
make go template working

andig commented on templates/definition/meter/senec-home.yaml:
```diff
@@ -17,6 +17,12 @@ params:
       - http
       - https
     default: http
+  - name: live_stats
```
Could we make this `livestats`? Think we typically don't use underscores. Could also rename this to `version` with v3 and v2 as values? This could be useful for later extensions.

premultiply commented:
First of all this needs a verification on the data of the different models and firmware.
Easiest way is to drop energy at all again.

andig commented:
Then lets do that- as-is Senec is now broken.

Soph committed:
switching to version

Soph commented:
I switched to version and added `v2`, `v3`, `v4` as options. In case of `v2` energy will not be used. Others have no effect but might be useful in the future.

andig committed:
wip

andig committed:
wip

andig commented on templates/definition/meter/senec-home.yaml:
```diff
@@ -43,6 +52,7 @@ render: |
     jq: .ENERGY.GUI_BAT_DATA_POWER | sub("fl_"; "")
     scale: -1
   {{- end }}
+  {{- if ne .version "v2" }}
```
```suggestion
  {{- if eq .version "v3" }}
```

Müssen wir für Rückwärtskompatibilität umdrehen.

andig committed:
Update templates/definition/meter/senec-home.yaml

premultiply commented:
Nochmal: Ich möchte erst die Vergleichsdaten der aufgeführten Systeme gesehen haben. Dass es da diesen angenommen Zusammenhang gibt ist nicht belegt.

Hier irgendwelche Phantasienummern aufzuzählen nutzt niemandem etwas.

Ich werfe gleich nur in einem anderen PR die Energie wieder raus und dann ist das erstmal gut.

Bis die angefragten Daten (Gesamtauszug von lala.cgi) vorliegen mache ich hier erstmal zu.

Soph commented:
> Nochmal: Ich möchte erst die Vergleichsdaten der aufgeführten Systeme gesehen haben. Dass es da diesen angenommen Zusammenhang gibt ist nicht belegt.

Kann man da helfen, wird das irgendwo getrackt? Ich habe wie gesagt einen 2.1 zuhause.

> Hier irgendwelche Phantasienummern aufzuzählen nutzt niemandem etwas.

Da komme ich nicht ganz mit. https://senec.com/de/produkte/senec-home Es gibt Senec.Home 2.1, 3 und 4. Da ist `v2`, `v3` und `v4` jetzt nicht völlig aus der Luft gegriffen. Was sicherlich nicht 100% belegt ist, warum meine Speicher diese Daten nicht hat. Aber da er offensichtlich gerade die neueste Firmware bekommen hat (weil http ging nicht mehr) ist die Wahrscheinlichkeit, dass es an der Speicher Version liegt nicht ganz von der Hand zuweisen.

> Ich werfe gleich nur in einem anderen PR die Energie wieder raus und dann ist das erstmal gut.

Danke. Das aktuelle template im doc folder ist aber weiterhin nicht korrekt, da es `protocol` verwendet während das eigentlich template `schema` hat. Kann dazu gerne noch einen PR aufmachen. Bin mir aber jetzt unsicher ob das überhaupt gewünscht ist. 

> Bis die angefragten Daten (Gesamtauszug von lala.cgi) vorliegen mache ich hier erstmal zu.

Siehe oben, helfe gerne. Oder wo wurden die Daten angefragt?

Und dann nur mal so, ich kann euch null reinreden wie ihr dieses Projekt managt. Aber versetze dich doch einfach mal in meine Lage vielleicht kurz. Ich mache mir die Arbeit und mache nen PR auf und klar müsst ihr den nicht nehmen (habe ich gleich schon in der anderen issue geschrieben) aber lies doch vielleicht nochmal deinen Kommentar. Aber vermutlich gehe ich einfach mal davon aus, dass du mit @andig gesprochen hast und nicht mir.

neubi4 commented:
Moin,

hier alle Daten von meinem Senec v2.1 mit dem neusten Update (mit https), ich hoffe das ist das was du sehen willst @premultiply wenn noch was anderes gebraucht wird einfach bescheid sagen.

Geholt mit einem POST an die lala.cgi mit folgendem JSON Body:
<details>
<summary>Request JSON Body</summary>

```json
{
  "STATISTIC": {},
  "ENERGY": {},
  "FEATURES": {},
  "LOG": {},
  "SYS_UPDATE": {},
  "WIZARD": {},
  "BMS": {},
  "BAT1": {},
  "BAT1OBJ1": {},
  "BAT1OBJ2": {},
  "BAT1OBJ3": {},
  "BAT1OBJ4": {},
  "PWR_UNIT": {},
  "PV1": {}
}
```

</details>
<details>
<summary>Response</summary>


```json
{
    "STATISTIC": {},
    "ENERGY": {
        "CAPTESTMODULE": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "GUI_BAT_DATA_COLLECTED": "u8_01",
        "GUI_BAT_DATA_CURRENT": "fl_BE2E147B",
        "GUI_BAT_DATA_FUEL_CHARGE": "fl_428C0000",
        "GUI_BAT_DATA_MAX_CELL_VOLTAGE": "u1_0EE9",
        "GUI_BAT_DATA_MIN_CELL_VOLTAGE": "u1_0EE1",
        "GUI_BAT_DATA_POWER": "fl_C1113943",
        "GUI_BAT_DATA_VOLTAGE": "fl_42558F5C",
        "GUI_BOOSTING_INFO": "u8_00",
        "GUI_CAP_TEST_STATE": "u8_00",
        "GUI_CHARGING_INFO": "u8_00",
        "GUI_GRID_POW": "fl_C5408FD7",
        "GUI_HOUSE_POW": "fl_4522A162",
        "GUI_INIT_CHARGE_START": "u8_00",
        "GUI_INIT_CHARGE_STOP": "u8_00",
        "GUI_INVERTER_POWER": "fl_45B15000",
        "GUI_TEST_CHARGE_STAT": "u8_00",
        "GUI_TEST_DISCHARGE_STAT": "u8_00",
        "INIT_CHARGE_ACK": "u8_00",
        "INIT_CHARGE_DIFF_VOLTAGE": "fl_00000000",
        "INIT_CHARGE_MAX_CURRENT": "fl_00000000",
        "INIT_CHARGE_MAX_VOLTAGE": "fl_00000000",
        "INIT_CHARGE_MIN_VOLTAGE": "fl_00000000",
        "INIT_CHARGE_RERUN": "u8_00",
        "INIT_CHARGE_RUNNING": "u8_00",
        "INIT_CHARGE_STATE": "u8_00",
        "INIT_CHARGE_TIMER": "u6_0000000000000000",
        "INIT_DISCHARGE_MAX_CURRENT": "fl_00000000",
        "LI_STORAGE_MODE_RUNNING": "u8_00",
        "LI_STORAGE_MODE_START": "u8_00",
        "LI_STORAGE_MODE_STOP": "u8_00",
        "SAFE_CHARGE_FORCE": "u8_00",
        "SAFE_CHARGE_PROHIBIT": "u8_00",
        "SAFE_CHARGE_RUNNING": "u8_00",
        "STAT_HOURS_OF_OPERATION": "u3_00003575",
        "STAT_LIMITED_NET_SKEW": "u8_00",
        "STAT_STATE": "u8_0D",
        "ZERO_EXPORT": "u8_00"
    },
    "FEATURES": {
        "CAR": "u8_00",
        "CLOUDREADY": "u8_01",
        "ECOGRIDREADY": "u8_01",
        "HEAT": "u8_00",
        "ISLAND": "u8_00",
        "ISLAND_PRO": "u8_00",
        "PEAKSHAVING": "u8_00",
        "SGREADY": "u8_00",
        "SHKW": "u8_00",
        "SOCKETS": "u8_00"
    },
    "LOG": {
        "LOG_IN_BUTT": "u8_00",
        "LOG_IN_NOK_COUNT": "u8_00",
        "LOG_OUT_BUTT": "u8_00",
        "PASSWORD": "st_",
        "USERNAME": "st_",
        "USER_LEVEL": "u8_00"
    },
    "SYS_UPDATE": {
        "FSM_STATE": "u8_01",
        "MISC": [
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "NPU_IMAGE_VERSION": "u1_0905",
        "NPU_VER": "u1_000A",
        "UPDATE_AVAILABLE": "u8_00",
        "USER_REBOOT_DEVICE": "u8_00",
        "USER_REQ_UPDATE": "u8_00"
    },
    "WIZARD": {
        "APPLICATION_HASH": "st_478e9310f37c4f8adafd861cc723904796885136",
        "APPLICATION_VERSION": "st_0825",
        "BATT_IPU_MISMATCH": "u8_00",
        "BOOT": "u8_01",
        "CHARGE_PRIO": "u8_00",
        "CONFIG_CHECKSUM": "u1_98EE",
        "CONFIG_LOADED": "u8_01",
        "CONFIG_MODIFIED_BY_USER": "u8_00",
        "CONFIG_WRITE": "u8_00",
        "DEVICE_BATTERY_TYPE": "u8_03",
        "DEVICE_INVERTER_TYPE": "u8_42",
        "DEVICE_INV_ENABLED": [
            "u8_01",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "DEVICE_INV_PHASES_ARR": [
            "u8_04",
            "u8_04",
            "u8_04",
            "u8_04",
            "u8_04",
            "u8_04"
        ],
        "DEVICE_INV_SLAVE_ADRESS": [
            "u8_01",
            "u8_04",
            "u8_05",
            "u8_06",
            "u8_07",
            "u8_08"
        ],
        "DEVICE_PM_GRID_ENABLED": "u8_01",
        "DEVICE_PM_HOUSE_ENABLED": "u8_00",
        "DEVICE_PM_TYPE": "u8_01",
        "DEVICE_WB_TYPE": "u8_00",
        "FEATURECODE_ENTERED": "u8_00",
        "FIRMWARE_VERSION": "st_2307",
        "GRID_CONNECTION_TYPE": "u8_02",
        "GUI_LANG": "u8_00",
        "HEAT_CONN_TYPE": "u8_02",
        "INSULATION_RESISTANCE": "u3_000003E8",
        "INTERFACE_VERSION": "st_0966",
        "LOGGER_SEVERITY": "u8_08",
        "MAC_ADDRESS_BYTES": [
            "u8_14",
            "u8_1F",
            "u8_BA",
            "u8_21",
            "u8_3F",
            "u8_67"
        ],
        "MASTER_SLAVE_ADDRESSES": [
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "MASTER_SLAVE_MODE": "u8_00",
        "PS_ENABLE": "u8_00",
        "PS_HOUR": "u8_00",
        "PS_MINUTE": "u8_00",
        "PS_RESERVOIR": "u8_00",
        "PV_CONFIG": [
            "u8_01",
            "u8_01"
        ],
        "PWRCFG_PEAK_PV_POWER": "fl_46091C00",
        "SETUP_ABS_POWER": "u3_00000000",
        "SETUP_HV_PHASE": "u8_00",
        "SETUP_NUMBER_WALLBOXES": "u8_00",
        "SETUP_PM_GRID_ADR": "u8_01",
        "SETUP_PM_HOUSE_ADR": "u8_02",
        "SETUP_POWER_RULE": "u8_46",
        "SETUP_PV_INV_IP0": "u3_00000000",
        "SETUP_PV_INV_IP1": "u3_00000000",
        "SETUP_PV_INV_IP2": "u3_00000000",
        "SETUP_PV_INV_IP3": "u3_00000000",
        "SETUP_PV_INV_IP4": "u3_00000000",
        "SETUP_PV_INV_IP5": "u3_00000000",
        "SETUP_RCR_STEPS": [
            "u1_0000",
            "u1_001E",
            "u1_003C",
            "u1_0064"
        ],
        "SETUP_USED_PHASE": "u8_01",
        "SETUP_USE_ABS_POWER": "u8_00",
        "SETUP_USE_DRM0": "u8_00",
        "SETUP_USE_RCR": "u8_00",
        "SETUP_WALLBOX_SERIAL0": "st_",
        "SETUP_WALLBOX_SERIAL1": "st_",
        "SETUP_WALLBOX_SERIAL2": "st_",
        "SETUP_WALLBOX_SERIAL3": "st_",
        "SG_READY_CURR_MODE": "u8_01",
        "SG_READY_ENABLED": "u8_00",
        "SG_READY_ENABLE_OVERWRITE": "u8_00",
        "SG_READY_EN_MODE1": "u8_00",
        "SG_READY_OVERWRITE_RELAY": [
            "u8_00",
            "u8_00"
        ],
        "SG_READY_POWER_COMM": "u1_FFFF",
        "SG_READY_POWER_NORMAL": "u1_0000",
        "SG_READY_POWER_PROP": "u1_FFFF",
        "SG_READY_TIME": "u1_02D0",
        "ZEROMODULE": "u8_00"
    },
    "BMS": {
        "ALARM_STATUS": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "BATTERY_STATUS": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "BL": [
            "u1_30C0",
            "u1_30C0",
            "u1_30C0",
            "u1_0000"
        ],
        "BMS_READY_FLAG": "u8_01",
        "BMS_STATUS": "u8_01",
        "BMS_STATUS_TIMESTAMP": "i3_64EF4239",
        "CELL_BALANCE_STATUS": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "CELL_TEMPERATURES_MODULE_A": [
            "i8_21",
            "i8_21",
            "i8_23",
            "i8_22",
            "i8_23",
            "i8_22"
        ],
        "CELL_TEMPERATURES_MODULE_B": [
            "i8_21",
            "i8_22",
            "i8_23",
            "i8_23",
            "i8_23",
            "i8_22"
        ],
        "CELL_TEMPERATURES_MODULE_C": [
            "i8_22",
            "i8_23",
            "i8_23",
            "i8_23",
            "i8_24",
            "i8_23"
        ],
        "CELL_TEMPERATURES_MODULE_D": [
            "i8_00",
            "i8_00",
            "i8_00",
            "i8_00",
            "i8_00",
            "i8_00"
        ],
        "CELL_VOLTAGES_MODULE_A": [
            "u1_0EE6",
            "u1_0EE6",
            "u1_0EE7",
            "u1_0EE4",
            "u1_0EE7",
            "u1_0EE4",
            "u1_0EE6",
            "u1_0EE5",
            "u1_0EE5",
            "u1_0EE6",
            "u1_0EE5",
            "u1_0EE5",
            "u1_0EE6",
            "u1_0EE8"
        ],
        "CELL_VOLTAGES_MODULE_B": [
            "u1_0EE5",
            "u1_0EE5",
            "u1_0EE6",
            "u1_0EE6",
            "u1_0EE7",
            "u1_0EE6",
            "u1_0EE5",
            "u1_0EE2",
            "u1_0EE4",
            "u1_0EE5",
            "u1_0EE4",
            "u1_0EE6",
            "u1_0EE6",
            "u1_0EE7"
        ],
        "CELL_VOLTAGES_MODULE_C": [
            "u1_0EE9",
            "u1_0EE7",
            "u1_0EE8",
            "u1_0EE6",
            "u1_0EE7",
            "u1_0EE7",
            "u1_0EE5",
            "u1_0EE1",
            "u1_0EE4",
            "u1_0EE7",
            "u1_0EE6",
            "u1_0EE7",
            "u1_0EE1",
            "u1_0EE7"
        ],
        "CELL_VOLTAGES_MODULE_D": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "CHARGED_ENERGY": [
            "u3_1F9B3133",
            "u3_1F9DD6F3",
            "u3_1F908EBB",
            "u3_00000000"
        ],
        "CHARGE_CURRENT_LIMIT": [
            "fl_40C00000",
            "fl_40C00000",
            "fl_40C00000",
            "fl_00000000"
        ],
        "COMMERRCOUNT": "u1_0000",
        "CURRENT": [
            "fl_BD4CCCCD",
            "fl_BD4CCCCD",
            "fl_BD8F5C29",
            "fl_00000000"
        ],
        "CYCLES": [
            "u1_003A",
            "u1_003A",
            "u1_003A",
            "u1_0000"
        ],
        "DERATING": "u8_00",
        "DISCHARGED_ENERGY": [
            "u3_2338A9FE",
            "u3_2339B455",
            "u3_23669B5C",
            "u3_00000000"
        ],
        "DISCHARGE_CURRENT_LIMIT": [
            "fl_C1400000",
            "fl_C1400000",
            "fl_C1400000",
            "fl_00000000"
        ],
        "ERROR": "u8_00",
        "FAULTLINECOUNT": "u1_0002",
        "FW": [
            "u1_9AD5",
            "u1_9AD5",
            "u1_9AD5",
            "u1_0000"
        ],
        "HW_EXTENSION": [
            "u1_0002",
            "u1_0002",
            "u1_0002",
            "u1_0000"
        ],
        "HW_MAINBOARD": [
            "u1_0BBA",
            "u1_0BBA",
            "u1_0BBA",
            "u1_0000"
        ],
        "MANUFACTURER": "u8_02",
        "MAX_CELL_VOLTAGE": [
            "u1_0EE8",
            "u1_0EE7",
            "u1_0EE9",
            "u1_0000"
        ],
        "MAX_TEMP": "i1_0168",
        "MIN_CELL_VOLTAGE": [
            "u1_0EE4",
            "u1_0EE2",
            "u1_0EE1",
            "u1_0000"
        ],
        "MIN_TEMP": "i1_014A",
        "MODULES_CONFIGURED": "u8_03",
        "MODULE_COUNT": "u8_03",
        "NOM_CHARGEPOWER_MODULE": "fl_441C4000",
        "NOM_DISCHARGEPOWER_MODULE": "fl_449C4000",
        "NR_INSTALLED": "u8_03",
        "PROTOCOL": "u1_0000",
        "RECOVERLOCKED": "u8_00",
        "SERIAL": [
            "st_",
            "st_",
            "st_",
            "st_"
        ],
        "SN": [
            "u3_029F7507",
            "u3_029F7509",
            "u3_029F7505",
            "u3_00000000"
        ],
        "SOC": [
            "u1_0043",
            "u1_0043",
            "u1_0043",
            "u1_0000"
        ],
        "SOH": [
            "u8_63",
            "u8_63",
            "u8_63",
            "u8_00"
        ],
        "STATUS": [
            "u3_00000001",
            "u3_00000001",
            "u3_00000001",
            "u3_00000000"
        ],
        "SYSTEM_SOC": "u1_029E",
        "TEMP_MAX": [
            "i8_23",
            "i8_23",
            "i8_24",
            "i8_00"
        ],
        "TEMP_MIN": [
            "i8_21",
            "i8_21",
            "i8_22",
            "i8_00"
        ],
        "TF_ERROR": "u8_00",
        "VOLTAGE": [
            "fl_42559168",
            "fl_42558B44",
            "fl_42559375",
            "fl_00000000"
        ],
        "WIZARD_ABORT": "u8_01",
        "WIZARD_CONFIRM": "u8_00",
        "WIZARD_DCCONNECT": "u1_0000",
        "WIZARD_START": "u8_00",
        "WIZARD_STATE": "u8_00"
    },
    "BAT1": {
        "CEI_LIMIT": "u1_0000",
        "DRM0_ASSERT": "u8_00",
        "ISLAND_ENABLE": "u8_00",
        "NSP2_FW": "u1_00F3",
        "NSP_FW": "u1_00F3",
        "RESET": "u8_00",
        "SELFTEST_ACT": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "SELFTEST_LIMIT": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "SELFTEST_OFF": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "SELFTEST_OVERALL_STATE": "u8_00",
        "SELFTEST_STATE": [
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "SELFTEST_STEP": [
            "st_",
            "st_",
            "st_",
            "st_",
            "st_",
            "st_",
            "st_",
            "st_"
        ],
        "SELFTEST_TIME": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "SERIAL": "st_",
        "SPARE_CAPACITY": "u8_00",
        "TRIG_ITALY_SELF": "u8_00",
        "TYPE": "u8_03"
    },
    "BAT1OBJ1": {
        "COMM": "u8_00",
        "I_DC": "fl_3DCCCCCD",
        "P": "i1_0000",
        "Q": "i1_0000",
        "S": "i1_0000",
        "SW_VERSION": "u3_00000294",
        "SW_VERSION2": "u3_000000F3",
        "SW_VERSION3": "u3_000000F3",
        "TEMP1": "u1_001F",
        "TEMP2": "u1_001F",
        "TEMP3": "u1_0024",
        "TEMP4": "u1_0000",
        "TEMP5": "u1_0000",
        "U_DC": "fl_42540000"
    },
    "BAT1OBJ2": {
        "OBJECT_NOT_FOUND": ""
    },
    "BAT1OBJ3": {
        "OBJECT_NOT_FOUND": ""
    },
    "BAT1OBJ4": {
        "OBJECT_NOT_FOUND": ""
    },
    "PWR_UNIT": {
        "ADRESS": [
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "CONNPWR": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CONNPWR_1": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CONNPWR_2": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CONNPWR_3": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CURRENTTEMP_MAX": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CURRENTTEMP_MAX_HW": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CURRENTTEMP_MIN": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "CURRENTTEMP_MIN_HW": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "ENFLURI": [
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "FW_VER": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "HW_REV": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "POWER": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "POWER_L1": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "POWER_L2": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "POWER_L3": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "PU_MISSING": "u8_01",
        "REQ_POWER": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "SERIAL": [
            "st_",
            "st_",
            "st_",
            "st_",
            "st_",
            "st_"
        ],
        "STATUS": [
            "u3_00000000",
            "u3_00000001",
            "u3_00000002",
            "u3_00000003",
            "u3_00000004",
            "u3_00000005"
        ],
        "TEMPMAX": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "TEMPMIN": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "TEMPTARGET": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "TEMP_COUNT": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ],
        "TEMP_LIMIT_LOWER": "fl_00000000",
        "TEMP_LIMIT_UPPER": "fl_42C80000",
        "TYPE": [
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00",
            "u8_00"
        ],
        "WATERVOL": [
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000",
            "u1_0000"
        ]
    },
    "PV1": {
        "ERROR_STATE_INT": [
            "u3_00000000",
            "u3_00000000"
        ],
        "INTERNAL_INV_ERROR_TEXT": [
            "st_",
            "st_"
        ],
        "INTERNAL_INV_ERR_STATE_VALID": [
            "u8_00",
            "u8_00"
        ],
        "INTERNAL_INV_STATE": [
            "u8_05",
            "u8_05"
        ],
        "INTERNAL_MD_AVAIL": [
            "u8_00",
            "u8_00"
        ],
        "INTERNAL_MD_MANUFACTURER": [
            "st_",
            "st_"
        ],
        "INTERNAL_MD_MODEL": [
            "st_",
            "st_"
        ],
        "INTERNAL_MD_SERIAL": [
            "st_",
            "st_"
        ],
        "INTERNAL_MD_VERSION": [
            "st_",
            "st_"
        ],
        "INTERNAL_PV_AVAIL": "u8_00",
        "INV_MODEL": [
            "st_",
            "st_END_OF_ARRAY"
        ],
        "INV_SERIAL": [
            "st_",
            "st_END_OF_ARRAY"
        ],
        "INV_VERSIONS": [
            "st_",
            "st_END_OF_ARRAY"
        ],
        "MPP_AVAIL": "u8_00",
        "MPP_CUR": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "MPP_POWER": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "MPP_VOL": [
            "fl_00000000",
            "fl_00000000",
            "fl_00000000"
        ],
        "POWER_RATIO": "fl_42C80000",
        "POWER_RATIO_L1": "fl_42C80000",
        "POWER_RATIO_L2": "fl_42C80000",
        "POWER_RATIO_L3": "fl_42C80000",
        "PV_MISSING": "u8_00",
        "P_TOTAL": "fl_C5B15000",
        "STATE_INT": [
            "u3_00000000",
            "u3_00000000"
        ],
        "TYPE": "u8_42"
    }
}
```
</details>

andig commented:
@premultiply weiter auf der Basis? Brauchst Du noch was?

premultiply commented:
Ja, v3 zum Vergleich.

dscho commented on templates/definition/meter/senec-home.yaml:
```diff
@@ -17,6 +17,15 @@ params:
     default: http
+  - name: version
+    description:
+      de: Senic.Home Version
+      en: Senic.Home Version
```
```suggestion
      de: Senec.Home Version
      en: Senec.Home Version
```

dscho commented:
> Ja, v3 zum Vergleich.

@premultiply so kurz und knackig das auch ist, das ist vielleicht etwas... _zu_ kurz und knackig. Kannst Du mal einen konkreten `curl` Aufruf 'raushauen, den ich hier absetzen kann? Ich habe einen (gedrosselten) SENEC.Home v3 hybrid duo hier.

neubi4 commented:
> > Ja, v3 zum Vergleich.
> 
> @premultiply so kurz und knackig das auch ist, das ist vielleicht etwas... _zu_ kurz und knackig. Kannst Du mal einen konkreten `curl` Aufruf 'raushauen, den ich hier absetzen kann? Ich habe einen (gedrosselten) SENEC.Home v3 hybrid duo hier.

So hier:

`curl -k https://senec-ip/lala.cgi -X POST -d '{"STATISTIC":{},"ENERGY":{},"FEATURES":{},"LOG":{},"SYS_UPDATE":{},"WIZARD":{},"BMS":{},"BAT1":{},"BAT1OBJ1":{},"BAT1OBJ2":{},"BAT1OBJ3":{},"BAT1OBJ4":{},"PWR_UNIT":{},"PV1":{}}'`

dscho commented:
> curl -k https://senec-ip/lala.cgi -X POST -d '{"STATISTIC":{},"ENERGY":{},"FEATURES":{},"LOG":{},"SYS_UPDATE":{},"WIZARD":{},"BMS":{},"BAT1":{},"BAT1OBJ1":{},"BAT1OBJ2":{},"BAT1OBJ3":{},"BAT1OBJ4":{},"PWR_UNIT":{},"PV1":{}}'

Sehr schön, danke Dir! Mein SENEC.Home v3 hybrid duo spricht noch nicht https, aber mit http kriege ich das hier:

```json
{
   "BAT1" : {
      "CEI_LIMIT" : "u1_0000",
      "DRM0_ASSERT" : "u8_00",
      "ISLAND_ENABLE" : "u8_00",
      "NSP2_FW" : "u1_0006",
      "NSP_FW" : "u1_000A",
      "RESET" : "u8_00",
      "SELFTEST_ACT" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "SELFTEST_LIMIT" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "SELFTEST_OFF" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "SELFTEST_OVERALL_STATE" : "u8_05",
      "SELFTEST_STATE" : [
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "SELFTEST_STEP" : [
         "st_",
         "st_",
         "st_",
         "st_",
         "st_",
         "st_",
         "st_",
         "st_"
      ],
      "SELFTEST_TIME" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "SERIAL" : "st_771078IB008954690004",
      "SPARE_CAPACITY" : "u8_00",
      "TRIG_ITALY_SELF" : "u8_00",
      "TYPE" : "u8_04"
   },
   "BAT1OBJ1" : {
      "COMM" : "u8_00",
      "I_DC" : "fl_C20E199A",
      "P" : "i1_0000",
      "Q" : "i1_0000",
      "S" : "i1_0000",
      "SW_VERSION" : "u3_03100014",
      "SW_VERSION2" : "u3_0403000A",
      "SW_VERSION3" : "u3_05040006",
      "TEMP1" : "u1_0024",
      "TEMP2" : "u1_002C",
      "TEMP3" : "u1_0000",
      "TEMP4" : "u1_0000",
      "TEMP5" : "u1_0000",
      "U_DC" : "fl_42427AE1"
   },
   "BAT1OBJ2" : {
      "OBJECT_NOT_FOUND" : ""
   },
   "BAT1OBJ3" : {
      "OBJECT_NOT_FOUND" : ""
   },
   "BAT1OBJ4" : {
      "OBJECT_NOT_FOUND" : ""
   },
   "BMS" : {
      "ALARM_STATUS" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "BATTERY_STATUS" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "BL" : [
         "u1_30C0",
         "u1_30C0",
         "u1_30C0",
         "u1_0000"
      ],
      "BMS_READY_FLAG" : "u8_01",
      "BMS_STATUS" : "u8_01",
      "BMS_STATUS_TIMESTAMP" : "i3_64D64136",
      "CELL_BALANCE_STATUS" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "CELL_TEMPERATURES_MODULE_A" : [
         "i8_18",
         "i8_18",
         "i8_19",
         "i8_19",
         "i8_19",
         "i8_19"
      ],
      "CELL_TEMPERATURES_MODULE_B" : [
         "i8_18",
         "i8_18",
         "i8_19",
         "i8_19",
         "i8_19",
         "i8_19"
      ],
      "CELL_TEMPERATURES_MODULE_C" : [
         "i8_19",
         "i8_19",
         "i8_19",
         "i8_19",
         "i8_1A",
         "i8_19"
      ],
      "CELL_TEMPERATURES_MODULE_D" : [
         "i8_00",
         "i8_00",
         "i8_00",
         "i8_00",
         "i8_00",
         "i8_00"
      ],
      "CELL_VOLTAGES_MODULE_A" : [
         "u1_0DAB",
         "u1_0DAF",
         "u1_0DAF",
         "u1_0DAF",
         "u1_0DAF",
         "u1_0DAF",
         "u1_0DB1",
         "u1_0DA4",
         "u1_0DB0",
         "u1_0DB0",
         "u1_0DAF",
         "u1_0DB0",
         "u1_0DB0",
         "u1_0DAC"
      ],
      "CELL_VOLTAGES_MODULE_B" : [
         "u1_0DA8",
         "u1_0DAA",
         "u1_0DAA",
         "u1_0DAB",
         "u1_0DAC",
         "u1_0DAC",
         "u1_0DAC",
         "u1_0D9F",
         "u1_0DAC",
         "u1_0DAB",
         "u1_0DAB",
         "u1_0DAB",
         "u1_0DAB",
         "u1_0DA9"
      ],
      "CELL_VOLTAGES_MODULE_C" : [
         "u1_0DAA",
         "u1_0DAC",
         "u1_0DAD",
         "u1_0DAE",
         "u1_0DAF",
         "u1_0DAE",
         "u1_0DB0",
         "u1_0DA1",
         "u1_0DAE",
         "u1_0DAD",
         "u1_0DAD",
         "u1_0DAD",
         "u1_0DAD",
         "u1_0DAA"
      ],
      "CELL_VOLTAGES_MODULE_D" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "CHARGED_ENERGY" : [
         "u3_05B86A5E",
         "u3_05B1FD7E",
         "u3_05B531C3",
         "u3_00000000"
      ],
      "CHARGE_CURRENT_LIMIT" : [
         "fl_41400000",
         "fl_41400000",
         "fl_41400000",
         "fl_00000000"
      ],
      "COMMERRCOUNT" : "u1_0000",
      "CURRENT" : [
         "fl_C14051EC",
         "fl_C13E8F5C",
         "fl_C14147AE",
         "fl_00000000"
      ],
      "CYCLES" : [
         "u1_000B",
         "u1_000B",
         "u1_000B",
         "u1_0000"
      ],
      "DERATING" : "u8_00",
      "DISCHARGED_ENERGY" : [
         "u3_0557A152",
         "u3_055F1C2A",
         "u3_056EC9F5",
         "u3_00000000"
      ],
      "DISCHARGE_CURRENT_LIMIT" : [
         "fl_C1C00000",
         "fl_C1C00000",
         "fl_C1C00000",
         "fl_00000000"
      ],
      "ERROR" : "u8_00",
      "FAULTLINECOUNT" : "u1_0001",
      "FW" : [
         "u1_75B6",
         "u1_75B6",
         "u1_75B6",
         "u1_0000"
      ],
      "HW_EXTENSION" : [
         "u1_0064",
         "u1_0064",
         "u1_0064",
         "u1_0000"
      ],
      "HW_MAINBOARD" : [
         "u1_0C1D",
         "u1_0C1D",
         "u1_0C1D",
         "u1_0000"
      ],
      "MANUFACTURER" : "u8_02",
      "MAX_CELL_VOLTAGE" : [
         "u1_0DB1",
         "u1_0DAC",
         "u1_0DAF",
         "u1_0000"
      ],
      "MAX_TEMP" : "i1_0104",
      "MIN_CELL_VOLTAGE" : [
         "u1_0DA4",
         "u1_0D9F",
         "u1_0DA1",
         "u1_0000"
      ],
      "MIN_TEMP" : "i1_00F0",
      "MODULES_CONFIGURED" : "u8_03",
      "MODULE_COUNT" : "u8_03",
      "NOM_CHARGEPOWER_MODULE" : "fl_441C4000",
      "NOM_DISCHARGEPOWER_MODULE" : "fl_449C4000",
      "NR_INSTALLED" : "u8_03",
      "PROTOCOL" : "u1_0000",
      "RECOVERLOCKED" : "u8_00",
      "SERIAL" : [
         "st_",
         "st_",
         "st_",
         "st_"
      ],
      "SN" : [
         "u3_02DCA88B",
         "u3_02DCA87F",
         "u3_02DCA889",
         "u3_00000000"
      ],
      "SOC" : [
         "u1_001C",
         "u1_001B",
         "u1_001B",
         "u1_0000"
      ],
      "SOH" : [
         "u8_63",
         "u8_63",
         "u8_63",
         "u8_00"
      ],
      "STATUS" : [
         "u3_00000001",
         "u3_00000001",
         "u3_00000001",
         "u3_00000000"
      ],
      "SYSTEM_SOC" : "u1_010E",
      "TEMP_MAX" : [
         "i8_19",
         "i8_19",
         "i8_1A",
         "i8_00"
      ],
      "TEMP_MIN" : [
         "i8_18",
         "i8_18",
         "i8_19",
         "i8_00"
      ],
      "TF_ERROR" : "u8_00",
      "VOLTAGE" : [
         "fl_42441EB8",
         "fl_4243E24E",
         "fl_42440312",
         "fl_00000000"
      ],
      "WIZARD_ABORT" : "u8_01",
      "WIZARD_CONFIRM" : "u8_00",
      "WIZARD_DCCONNECT" : "u1_0000",
      "WIZARD_START" : "u8_00",
      "WIZARD_STATE" : "u8_00"
   },
   "ENERGY" : {
      "CAPTESTMODULE" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "GUI_BAT_DATA_COLLECTED" : "u8_01",
      "GUI_BAT_DATA_CURRENT" : "fl_C2100A3D",
      "GUI_BAT_DATA_FUEL_CHARGE" : "fl_41D219DB",
      "GUI_BAT_DATA_MAX_CELL_VOLTAGE" : "u1_0DB1",
      "GUI_BAT_DATA_MIN_CELL_VOLTAGE" : "u1_0D9F",
      "GUI_BAT_DATA_POWER" : "fl_C4DC9153",
      "GUI_BAT_DATA_VOLTAGE" : "fl_42440106",
      "GUI_BOOSTING_INFO" : "u8_01",
      "GUI_CAP_TEST_STATE" : "u8_00",
      "GUI_CHARGING_INFO" : "u8_00",
      "GUI_GRID_POW" : "fl_C237530E",
      "GUI_HOUSE_POW" : "fl_44FD805D",
      "GUI_INIT_CHARGE_START" : "u8_00",
      "GUI_INIT_CHARGE_STOP" : "u8_00",
      "GUI_INVERTER_POWER" : "fl_439AA688",
      "GUI_TEST_CHARGE_STAT" : "u8_00",
      "GUI_TEST_DISCHARGE_STAT" : "u8_00",
      "INIT_CHARGE_ACK" : "u8_00",
      "INIT_CHARGE_DIFF_VOLTAGE" : "fl_00000000",
      "INIT_CHARGE_MAX_CURRENT" : "fl_00000000",
      "INIT_CHARGE_MAX_VOLTAGE" : "fl_00000000",
      "INIT_CHARGE_MIN_VOLTAGE" : "fl_00000000",
      "INIT_CHARGE_RERUN" : "u8_00",
      "INIT_CHARGE_RUNNING" : "u8_00",
      "INIT_CHARGE_STATE" : "u8_00",
      "INIT_CHARGE_TIMER" : "u6_0000000000000000",
      "INIT_DISCHARGE_MAX_CURRENT" : "fl_00000000",
      "LI_STORAGE_MODE_RUNNING" : "u8_00",
      "LI_STORAGE_MODE_START" : "u8_00",
      "LI_STORAGE_MODE_STOP" : "u8_00",
      "OFFPEAK_DURATION" : "u1_0000",
      "OFFPEAK_POWER" : "fl_00000000",
      "OFFPEAK_RUNNING" : "u8_00",
      "OFFPEAK_TARGET" : "u8_64",
      "SAFE_CHARGE_FORCE" : "u8_00",
      "SAFE_CHARGE_PROHIBIT" : "u8_00",
      "SAFE_CHARGE_RUNNING" : "u8_00",
      "STAT_HOURS_OF_OPERATION" : "u3_000008B9",
      "STAT_LIMITED_NET_SKEW" : "u8_00",
      "STAT_STATE" : "u8_11",
      "VPP_ACTIVATE_EXPORT_LIMIT" : "u8_00",
      "VPP_ACTIVATE_TARGET_POWER" : "u8_00",
      "VPP_DAILY_PARAMETER_RESET" : "u8_01",
      "VPP_ENDTIME_HOUR" : "u8_00",
      "VPP_ENDTIME_MINUTE" : "u8_00",
      "VPP_EXPORT_LIMIT" : "u3_00000000",
      "VPP_IGNORE_COUNTRY_TYPE" : "u8_00",
      "VPP_IS_ACTIVE" : "u8_00",
      "VPP_LAST_CHANGE_UTC" : "u3_00000000",
      "VPP_STARTTIME_HOUR" : "u8_00",
      "VPP_STARTTIME_MINUTE" : "u8_00",
      "VPP_TARGET_POWER" : "fl_00000000",
      "ZERO_EXPORT" : "u8_00"
   },
   "FEATURES" : {
      "CAR" : "u8_01",
      "CLOUDREADY" : "u8_01",
      "ECOGRIDREADY" : "u8_01",
      "HEAT" : "u8_01",
      "ISLAND" : "u8_00",
      "ISLAND_PRO" : "u8_00",
      "PEAKSHAVING" : "u8_01",
      "SGREADY" : "u8_01",
      "SHKW" : "u8_00",
      "SOCKETS" : "u8_00"
   },
   "LOG" : {
      "LOG_IN_BUTT" : "u8_00",
      "LOG_OUT_BUTT" : "u8_00",
      "PASSWORD" : "st_",
      "USERNAME" : "st_",
      "USER_LEVEL" : "u8_00"
   },
   "PV1" : {
      "ERROR_STATE_INT" : [
         "u3_00000000",
         "u3_00000000"
      ],
      "INTERNAL_INV_ERROR_TEXT" : [
         "st_",
         "st_"
      ],
      "INTERNAL_INV_ERR_STATE_VALID" : [
         "u8_01",
         "u8_01"
      ],
      "INTERNAL_INV_STATE" : [
         "u8_02",
         "u8_02"
      ],
      "INTERNAL_MD_AVAIL" : [
         "u8_01",
         "u8_01"
      ],
      "INTERNAL_MD_MANUFACTURER" : [
         "st_Senec",
         "st_Senec"
      ],
      "INTERNAL_MD_MODEL" : [
         "st_V3 LV",
         "st_V3 HV"
      ],
      "INTERNAL_MD_SERIAL" : [
         "st_771078IB008954690004",
         "st_765856IB008937980004"
      ],
      "INTERNAL_MD_VERSION" : [
         "st_HMI: 3.16.20 PU: 4.3.10 BDC: 5.4.6",
         "st_HMI: 3.16.20 PU: 4.3.10"
      ],
      "INTERNAL_PV_AVAIL" : "u8_02",
      "INV_MODEL" : [
         "st_",
         "st_",
         "st_END_OF_ARRAY"
      ],
      "INV_SERIAL" : [
         "st_",
         "st_",
         "st_END_OF_ARRAY"
      ],
      "INV_VERSIONS" : [
         "st_HMI: 3.16.20 PU: 4.3.10",
         "st_HMI: 3.16.20 PU: 4.3.10",
         "st_END_OF_ARRAY"
      ],
      "MPP_AVAIL" : "u8_03",
      "MPP_CUR" : [
         "fl_3EC624DE",
         "fl_3EE66667",
         "fl_3EC6A7F0"
      ],
      "MPP_POWER" : [
         "fl_42B0CF5D",
         "fl_42CD24DE",
         "fl_42EB4313"
      ],
      "MPP_VOL" : [
         "fl_436A26A9",
         "fl_4369E042",
         "fl_439D9E57"
      ],
      "POWER_RATIO" : "fl_42C80000",
      "POWER_RATIO_L1" : "fl_42C80000",
      "POWER_RATIO_L2" : "fl_42C80000",
      "POWER_RATIO_L3" : "fl_42C80000",
      "PV_MISSING" : "u8_00",
      "P_TOTAL" : "fl_C39A4DD4",
      "STATE_INT" : [
         "u3_00000110",
         "u3_00000110"
      ],
      "TYPE" : "u8_A0"
   },
   "PWR_UNIT" : {
      "ADRESS" : [
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "CONNPWR" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CONNPWR_1" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CONNPWR_2" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CONNPWR_3" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CURRENTTEMP_MAX" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CURRENTTEMP_MAX_HW" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CURRENTTEMP_MIN" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "CURRENTTEMP_MIN_HW" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "ENFLURI" : [
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "FW_VER" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "HW_REV" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "POWER" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "POWER_L1" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "POWER_L2" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "POWER_L3" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "PU_MISSING" : "u8_00",
      "REQ_POWER" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "SERIAL" : [
         "st_",
         "st_",
         "st_",
         "st_",
         "st_",
         "st_"
      ],
      "STATUS" : [
         "u3_00000000",
         "u3_00000001",
         "u3_00000002",
         "u3_00000003",
         "u3_00000004",
         "u3_00000005"
      ],
      "TEMPMAX" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "TEMPMIN" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "TEMPTARGET" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "TEMP_COUNT" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ],
      "TEMP_LIMIT_LOWER" : "fl_00000000",
      "TEMP_LIMIT_UPPER" : "fl_42C80000",
      "TYPE" : [
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "WATERVOL" : [
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000",
         "u1_0000"
      ]
   },
   "STATISTIC" : {
      "CURRENT_STATE" : "u1_0011",
      "LIVE_BAT_CHARGE" : "fl_439FA150",
      "LIVE_BAT_CHARGE_MASTER" : "fl_439FA150",
      "LIVE_BAT_DISCHARGE" : "fl_43963320",
      "LIVE_BAT_DISCHARGE_MASTER" : "fl_43963320",
      "LIVE_GRID_EXPORT" : "fl_44E62710",
      "LIVE_GRID_IMPORT" : "fl_423E26C2",
      "LIVE_HOUSE_CONS" : "fl_4478E8E6",
      "LIVE_PU_ENERGY" : [
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000",
         "fl_00000000"
      ],
      "LIVE_PV_GEN" : "fl_452F7FE0",
      "LIVE_PV_GEN_MASTER" : "fl_00000000",
      "LIVE_WB_ENERGY" : [
         "u6_0000000000000000",
         "u6_0000000000000000",
         "u6_0000000000000000",
         "u6_0000000000000000"
      ],
      "MEASURE_TIME" : "i3_64F19B08",
      "STAT_SUM_E_PU" : [
         "u6_0000000000000000",
         "u6_0000000000000000",
         "u6_0000000000000000",
         "u6_0000000000000000",
         "u6_0000000000000000",
         "u6_0000000000000000"
      ]
   },
   "SYS_UPDATE" : {
      "FSM_STATE" : "u8_01",
      "MISC" : [
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "NPU_IMAGE_VERSION" : "u1_07DB",
      "NPU_VER" : "u1_0009",
      "UPDATE_AVAILABLE" : "u8_00",
      "USER_REBOOT_DEVICE" : "u8_00",
      "USER_REQ_UPDATE" : "u8_00"
   },
   "WIZARD" : {
      "APPLICATION_HASH" : "st_06feefa94aaec4a945f78837773a48497562caa8",
      "APPLICATION_VERSION" : "st_1824",
      "BATT_IPU_MISMATCH" : "u8_00",
      "BOOT" : "u8_01",
      "CHARGE_PRIO" : "u8_00",
      "CONFIG_CHECKSUM" : "u1_9978",
      "CONFIG_LOADED" : "u8_01",
      "CONFIG_MODIFIED_BY_USER" : "u8_00",
      "CONFIG_WRITE" : "u8_00",
      "DEVICE_BATTERY_TYPE" : "u8_04",
      "DEVICE_INVERTER_TYPE" : "u8_A0",
      "DEVICE_INV_ENABLED" : [
         "u8_01",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "DEVICE_INV_PHASES_ARR" : [
         "u8_04",
         "u8_04",
         "u8_04",
         "u8_04",
         "u8_04",
         "u8_04"
      ],
      "DEVICE_INV_SLAVE_ADRESS" : [
         "u8_03",
         "u8_04",
         "u8_05",
         "u8_06",
         "u8_07",
         "u8_08"
      ],
      "DEVICE_PM_GRID_ENABLED" : "u8_01",
      "DEVICE_PM_HOUSE_ENABLED" : "u8_00",
      "DEVICE_PM_TYPE" : "u8_02",
      "DEVICE_WB_TYPE" : "u8_00",
      "FEATURECODE_ENTERED" : "u8_00",
      "FIRMWARE_VERSION" : "st_954",
      "GENERATION_METER_SN" : "u3_00000000",
      "GRID_CONNECTION_TYPE" : "u8_02",
      "GUI_LANG" : "u8_00",
      "HEATPUMP_METER_SN" : "u3_00000000",
      "HEAT_CONN_TYPE" : "u8_02",
      "INSULATION_RESISTANCE" : "u3_000003E8",
      "INTERFACE_VERSION" : "st_0964",
      "LOGGER_SEVERITY" : "u8_08",
      "MAC_ADDRESS_BYTES" : [
         "u8_14",
         "u8_1F",
         "u8_BA",
         "u8_22",
         "u8_CE",
         "u8_92"
      ],
      "MASTER_SLAVE_ADDRESSES" : [
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00",
         "u8_00"
      ],
      "MASTER_SLAVE_MODE" : "u8_00",
      "POWER_METER_SERIAL" : "u3_00000000",
      "PS_ENABLE" : "u8_00",
      "PS_HOUR" : "u8_00",
      "PS_MINUTE" : "u8_00",
      "PS_RESERVOIR" : "u8_00",
      "PV_CONFIG" : [
         "u8_01",
         "u8_01"
      ],
      "PWRCFG_PEAK_PV_POWER" : "fl_46034000",
      "SENEC_METER_SN" : "u3_00000000",
      "SETUP_ABS_POWER" : "u3_00000000",
      "SETUP_HV_PHASE" : "u8_02",
      "SETUP_NUMBER_WALLBOXES" : "u8_00",
      "SETUP_PM_GRID_ADR" : "u8_01",
      "SETUP_PM_HOUSE_ADR" : "u8_02",
      "SETUP_POWER_RULE" : "u8_64",
      "SETUP_PV_INV_IP0" : "u3_00000000",
      "SETUP_PV_INV_IP1" : "u3_00000000",
      "SETUP_PV_INV_IP2" : "u3_00000000",
      "SETUP_PV_INV_IP3" : "u3_00000000",
      "SETUP_PV_INV_IP4" : "u3_00000000",
      "SETUP_PV_INV_IP5" : "u3_00000000",
      "SETUP_RCR_STEPS" : [
         "u1_0000",
         "u1_001E",
         "u1_003C",
         "u1_0064"
      ],
      "SETUP_USED_PHASE" : "u8_01",
      "SETUP_USE_ABS_POWER" : "u8_00",
      "SETUP_USE_DRM0" : "u8_00",
      "SETUP_USE_RCR" : "u8_00",
      "SETUP_WALLBOX_SERIAL0" : "st_",
      "SETUP_WALLBOX_SERIAL1" : "st_",
      "SETUP_WALLBOX_SERIAL2" : "st_",
      "SETUP_WALLBOX_SERIAL3" : "st_",
      "SG_READY_CURR_MODE" : "u8_01",
      "SG_READY_ENABLED" : "u8_00",
      "SG_READY_ENABLE_OVERWRITE" : "u8_00",
      "SG_READY_EN_MODE1" : "u8_00",
      "SG_READY_OVERWRITE_RELAY" : [
         "u8_00",
         "u8_00"
      ],
      "SG_READY_POWER_COMM" : "u1_FFFF",
      "SG_READY_POWER_PROP" : "u1_FFFF",
      "SG_READY_TIME" : "u1_02D0",
      "TEST_EG_METER" : "u8_00",
      "TEST_GENERATION_METER" : "u8_00",
      "TEST_HEATPUMP_METER" : "u8_00",
      "TEST_SENEC_METER" : "u8_00",
      "ZEROMODULE" : "u8_00"
   }
}
```

andig commented:
Zumindest scheint also https nicht an der Version zu hängen.

Commit:
Update templates/definition/meter/senec-home.yaml

Co-authored-by: Johannes Schindelin <dscho@github.com>

andig commented:
@premultiply hast Du schon eine Idee, wie wir hier weiter machen können?

premultiply commented:
Ich denke hier gibt es eigentlich nichts mehr zu tun.
Die wieder anderweitig entfernten Energiezähler haben das Problem schon gelöst.

Soph commented:
@premultiply mir soll es egal sein, ich hab nur einen v2, aber du hast das `Energy` doch nicht ohne Grund in dem PR hinzugefügt? Und nun ist es egal?
